### PR TITLE
Adding Sticky Card to AE2

### PR DESCRIPTION
**Pull request by Cardinalstars**
This addition is based on this issue: https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14787
I believe this can be a very useful addition to AE2, and solves some problems in a very elegant manner.

After this is merged in I will add similar functionality to AE2-Fluid and AE2-Essentia.

Any suggestions on tool-tips or other language stuff is welcome.

I also don't believe that what I've written here is the best way to do things. The AE2 codebase is quite dense, but I believe this is at least close to how the implementation should be done. Knowing this, any suggestions or requested changes to the code is greatly appreciated.

### CODE REVIEW

**Comment by Dream-Master** [comment]
@Cardinalstars you need to run spotless. Your build failed .

File: src/main/java/appeng/me/storage/VoidFluidInventory.java [hunk 1]
```diff
@@ -90,4 +90,9 @@ public int getSlot() {
     public boolean validForPass(final int i) {
         return i == 2;
     }
+
+    @Override
+    public boolean getSticky() {
```
Comment (Laiff) [inline src/main/java/appeng/me/storage/VoidFluidInventory.java:95]: If you extract it into separate interface like `HasSticky` with default implementation `() -> false` you can just add interface implementation without adding new method directly in each class

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java [hunk 2]
```diff
@@ -114,6 +118,10 @@ public void drawFG(final int offsetX, final int offsetY, final int mouseX, final
         if (this.rwMode != null) {
             this.rwMode.set(((ContainerStorageBus) this.cvb).getReadWriteMode());
         }
+
+        if (this.stickyMode != null) {
+            this.stickyMode.set(((ContainerStorageBus) this.cvb).getStickyMode());
```
Comment (Laiff) [inline src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:123]: It is better to check it via instanceof to be sure this not going into ClassCastException

Comment (Cardinalstars) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:123]: Should I change the other checks in the file as well to comply with this?

File: src/main/java/appeng/me/storage/VoidFluidInventory.java (hunk 1 above)
Comment (Cardinalstars) [reply src/main/java/appeng/me/storage/VoidFluidInventory.java:95]: To double check how this should be done do you want me to have this in an interface that is implemented by ONLY the classes that need it or should I leave it in IMEInventoryHandler.java and give it a default of false?

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java (hunk 2 above)
Comment (Laiff) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:123]: You can just check it using 
```java
if (this.cvb inctanceof ContainerStorageBus csb) {
   csb.getStickyMode()
}
```

File: src/main/java/appeng/api/storage/IMEInventoryHandler.java [hunk 3]
```diff
@@ -73,4 +73,13 @@ public interface IMEInventoryHandler<StackType extends IAEStack> extends IMEInve
...
     boolean validForPass(int i);
+
+    /**
+     * Gets whether an inventory is "Sticky" i.e. only it and other sticky storages that have partitions with certain
+     * items are allowed to be put into sticky storages.
+     * 
+     * @return
+     */
+    boolean getSticky();
+
```
Comment (Laiff) [inline src/main/java/appeng/api/storage/IMEInventoryHandler.java:84]:
```java
default boolean getSticky() {
    return false;
}
```

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java (hunk 2 above)
Comment (Cardinalstars) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:123]: I actually am going to remove this button as the sticky mode isn't controllable through a button, only a card. Forgot about that on accident.

Comment (Cardinalstars) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:123]: I went ahead and changed the other checks in this file to comply with your request.

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java
Comment (Cardinalstars) [inline src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:1]: @Laiff is this hat you were looking for basically? Is this check safer? I'm confused on how it is better, but I believe you.

Comment (Laiff) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:1]: Yes, but we need to preserve the logic for checking for null values in the fields.

Comment (Cardinalstars) [reply src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:1]: Null-checks are back in.

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java [hunk 4]
```diff
@@ -103,10 +103,16 @@ public void drawFG(final int offsetX, final int offsetY, final int mouseX, final
                 this.ySize - 96 + 3,
                 GuiColors.StorageBusInventory.getColor());
 
-        if (this.cvb instanceof ContainerStorageBus csb) {
-            this.storageFilter.set(csb.getStorageFilter());
-            this.rwMode.set(csb.getReadWriteMode());
-            this.fuzzyMode.set(csb.getFuzzyMode());
+        if (this.cvb instanceof ContainerStorageBus) {
```
Comment (Laiff) [inline src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:106]:
```suggestion
        if (this.cvb instanceof ContainerStorageBus cvb) {
```

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java [hunk 5]
```diff
@@ -103,10 +103,16 @@ public void drawFG(final int offsetX, final int offsetY, final int mouseX, final
...
+            if (this.fuzzyMode != null) {
+                this.fuzzyMode.set(this.cvb.getFuzzyMode());
+            }
+            if (this.storageFilter != null) {
+                this.storageFilter.set(((ContainerStorageBus) this.cvb).getStorageFilter());
```
Comment (Laiff) [inline src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:111]:
```suggestion
                this.storageFilter.set(cvb.getStorageFilter());
```

File: src/main/java/appeng/client/gui/implementations/GuiStorageBus.java [hunk 6]
```diff
@@ -103,10 +103,16 @@ public void drawFG(final int offsetX, final int offsetY, final int mouseX, final
...
+            }
+            if (this.rwMode != null) {
+                this.rwMode.set(((ContainerStorageBus) this.cvb).getReadWriteMode());
```
Comment (Laiff) [inline src/main/java/appeng/client/gui/implementations/GuiStorageBus.java:114]:
```suggestion
                this.rwMode.set(cvb.getReadWriteMode());
```

**Review by Laiff — COMMENTED** [review]
Mainly looks good but few small changes and we can try to roll it out on zeta

**Review by Laiff — APPROVED** [review]

**Pull request merged**